lib/internal/http2/compat.js

```diff
@@ -392,6 +409,16 @@ class Http2ServerResponse extends Stream {
     return '';
   }
 
+  set statusMessage(msg) {
+    if (statusMessageWarned === false) {
+      process.emitWarning(
+        'Status message is not supported by HTTP/2 (RFC7540 8.1.2.4)',
+        'UnsupportedWarning'
+      );
+      statusMessageWarned = true;
+    }
+  }
```

Comment:
this same block of code is shown in two other spots, can you refactor it to a single location? It needs a unit test as well.

Reply:
Definitely. Will update.